src/lib/statistics/statistics_objects/equal_distinct_count_histogram.hpp

```diff
@@ -1,17 +1,69 @@
 #pragma once
 
+#include <tsl/robin_map.h>  // NOLINT
 #include <memory>
 #include <string>
 #include <utility>
 #include <vector>
-
 #include "abstract_histogram.hpp"
 #include "types.hpp"
 
 namespace opossum {
 
 class Table;
 
+// Think of this as an unordered_map<T, HistogramCountType>. The hash, equals, and allocator template parameter are
```

Comment:
@Bensk1 we moved the `value_distribution` utility functions to the header, because now need them in the `equal_distinct_count_histogram.cpp` and the `top_k_as_generic_histogram.cpp`, and in this way can include them in both files. Leaving them in the `equal_distinct_count_histogram.cpp` resulted in including a `.cpp` file which isn't good style and the linter didn't like. 
Would it be better to only keep the method signatures in the `.hpp` and move the implementation to the `.cpp`?
We additionally discussed whether we should move the `value_distribution` utility functions to the `abstract_histogram` since both histograms inherit from it and already include its header. 
What would be your preferred option?

Reply:
Right, we are not going to include `.cpp` files. Also, I think that if this is shared behavior, it should not only be in one of the classes but either in `abstract_histogram` or in a `histogram_utils` file.

Reply:
Good choice.